sqlmesh/core/audit/builtin.py

```diff
@@ -59,7 +59,6 @@
     query="""
 SELECT 1
 FROM @this_model
-LIMIT @threshold + 1
```

Comment:
why was this needed to be removed? trino error?

Reply:
Ah yes I meant to make a comment on this. 

I don't understand how this query works. For example if I put this into Bigquery:
```
SELECT 1 FROM `sushi.items__fixture` HAVING COUNT(*) <= 100
```
I get this error:
```
The HAVING clause only allows aggregation if GROUP BY or SELECT list aggregation is present 
```
Which makes sense to me.

Reply:
The reason why I touched this at all is because Trino does not support the `+` in the limit. The more the read the query the more I was confused how this is supposed to work.

Reply:
ok change it to this

```
SELECT COUNT(*) AS cnt
FROM @this_model
LIMIT @threshold + 1
HAVING cnt <= @threshold
```

i will make sqlglot fix the limit @threshold

Reply:
```
SELECT COUNT(*) AS cnt
FROM @this_model
LIMIT @threshold + 1
HAVING cnt <= @threshold
```

The limit in this won't do anything. Limit is last in terms of order of operations and therefore since `count(*) will only return 1 row you are limiting the results of something that has a single row. 

If you want the limit on read then you need to do something like this:
```
SELECT COUNT(*) as cnt
FROM (
   SELECT 1
   FROM @this_model
   LIMIT @threshold + 1
)
HAVING cnt <= @threshold
```

Btw I don't think all engines support using an alias in having like that but not sure. It also might be solved by SQLGlot.

Reply:
i can do that, but i'm also not aware of an engine that doesn't support that

Reply:
let's do that subquery as you propose, i'm aware that bigquery has problems NOT using the alias (opposite of what you said)

Reply:
Ok will you update SQLGlot for Postgres to handle the limit issue?

Reply:
He updated here:  https://github.com/tobymao/sqlglot/commit/d9958f9438eecf8137361ab76f02acda63e59790